pkg/alertmanager/api.go

```diff
@@ -184,3 +185,42 @@ func validateUserConfig(logger log.Logger, cfg alertspb.AlertConfigDesc) error {
 
 	return nil
 }
+
+func (am *MultitenantAlertmanager) ListUserConfig(w http.ResponseWriter, r *http.Request) {
+	logger := util_log.WithContext(r.Context(), am.logger)
+	userIDs, err := am.store.ListAllUsers(r.Context())
+	if err != nil {
+		level.Error(logger).Log("msg", "failed to list users of alertmanager")
+		http.Error(w, fmt.Sprintf("%s: %s", errListAllUser, err.Error()), http.StatusInternalServerError)
+		return
+	}
+	userConfigMap := make(map[string]*UserConfig, len(userIDs))
+	for _, userID := range userIDs {
+		cfg, err := am.store.GetAlertConfig(r.Context(), userID)
+		if err != nil {
+			if err == alertspb.ErrNotFound {
+				http.Error(w, err.Error(), http.StatusNotFound)
```

Comment:
If config for single user is not found (perhaps it was just deleted), I don't think we should stop early... instead we should just ignore it, and return the rest. Note that `GetAlertConfigs` handles this.